stdlib/queue.mli

```diff
@@ -20,6 +20,99 @@
    {b Warning} This module is not thread-safe: each {!Queue.t} value
    must be protected from concurrent access (e.g. with a [Mutex.t]).
    Failure to do so can lead to a crash.
+
+   A basic example:
+    {[
+    # let q = Queue.create ();;
+    val q : '_weak1 Queue.t = <abstr>
+
+
+    # Queue.push 1 q; Queue.push 2 q; Queue.push 3 q;;
+    - : unit = ()
+
+    # Queue.length q;;
+    - : int = 3
+
+    # Queue.pop q;;
+    - : int = 1
+
+    # Queue.pop q;;
+    - : int = 2
+
+    # Queue.pop q;;
+    - : int = 3
+
+    # Queue.pop q;;
+    Exception: Stdlib.Queue.Empty.
+    ]}
+
+   For a more elaborate example, a classic algorithmic use of queues
+   is to implement a BFS (breadth-first search) through a graph.
+
+   {[
+     type graph = {
+       edges: (int, int list) Hashtbl.t
+     };;
+
+    (* Search in graph [g] using BFS, starting from node [start].
+       It returns the first node that satisfies [p], or [None] if
+       no node reachable from [start] satifies [p].
+    *)
+    let search_for ~(g:graph) ~(start:int) (p:int -> bool) : int option =
+      let to_explore = Queue.create() in
+      let explored = Hashtbl.create 16 in
+
+      Queue.push start to_explore;
```

Comment:
I'm surprised by the amount of empty lines in this example, it rather makes the code harder to read for me, but it's okay if it's your style / you have a strong preference.